client/widmgr/widmgr.go

```diff
@@ -187,13 +189,34 @@ func (m *WIDMgr) Shutdown() {
 
 // getIdentities fetches all signed identities or returns an error.
 func (m *WIDMgr) getIdentities() error {
+	m.lastTokenLock.Lock()
+	defer m.lastTokenLock.Unlock()
+
+	// initialize the map
+	m.lastToken = map[cstructs.TaskIdentity]*structs.SignedWorkloadIdentity{}
```

Comment:
Currently we wait to overwrite `m.lastToken` until after we've successfully called `SignIdentities`, so that we can keep the last tokens we saw if there's a transient problem (ex. the RPC connection was dropped briefly or something). So we should keep that behavior here and create this map as a local variable first and then assign it to `m.lastToken` after we've called `SignIdentities` below.

We could also avoid taking the lock until later if we did that.

Reply:
I think we could also create this map in `NewWIDMgr` since it has all the information it needs?